cpp/CMakeLists.txt

```diff
@@ -524,6 +527,7 @@ target_link_libraries(cudf
            PUBLIC ZLIB::ZLIB
                   ${ARROW_LIBRARIES}
                   cudf::Thrust
+                  cuco::cuco
```

Comment:
I should have been clearer, needs to be:

```
target_link_libraries(cudf
           PUBLIC ZLIB::ZLIB
                  ${ARROW_LIBRARIES}
                  cudf::Thrust
                  rmm::rmm
           PRIVATE cuco::cuco)
```

We aren't exposing any of the headers of cuco::cuco so it 
should be private as consumers of cudf will not need it.